### DISCUSSION TITLE
Why not remove the macro manager pane, simply treat a macro as a choice?

### DISCUSSION THREAD

**Discussion by zonegrave** [chhoumann/quickadd#762] [Ideas]
I'm currently using the quickAdd as a tool of auto generation, and it is very amazing and I love it.
But it seems a little bit confusing when I was using the Macro choice. You have to create a macro configure in the macro manager, and then create a macro choice,  bind to the macro configure you created and then you can use the macro choice to call the macro config. It seems a bit redundant, especially when I was trying to invoke a macro configure inside another macro, I have to create a macro choice for the invoked macro, and then invoke it as a choice inside the invoker macro.
I was wondering if we can just delete the macro manager part, and just treat a macro as a choice?

**Comment by zonegrave** [comment]
And I also came into the issue where I forgot to bind the correct macro configure to the macro choice I created. It is really annoying sometimes.

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

You were right, and this shipped in QuickAdd 2.0.0: the Macro Manager is gone. A Macro choice now contains its macro directly - you create the choice, click its gear, and build the command sequence right there. No separate manager, no binding step, no way to bind the wrong macro to the wrong choice.

For your invoke-a-macro-from-another-macro case: a macro can run any other choice via the "Choice" command type (or `executeChoice` from a script), so the dummy-choice dance is gone too.

Docs: https://quickadd.obsidian.guide/docs/Choices/MacroChoice

Closing as implemented - thanks for pushing on this.
